libs/core/algorithms/tests/unit/algorithms/util/test_simd_helpers.cpp

```diff
@@ -0,0 +1,105 @@
+//  Copyright (c) 2023 Johan511
+
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/unseq/simd_helpers.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <random>
+#include <utility>
+#include <vector>
+
+int seed = std::random_device{}();
+std::mt19937 gen(seed);
+
+using hpx::parallel::util::unseq_first_n, hpx::parallel::util::unseq2_first_n;
+
+template <typename T>
+void test_unseq_first_n1_dispatch2(std::size_t length, std::size_t first_index)
+{
+    first_index = first_index % length;
+
+    std::vector<T> v(length, static_cast<T>(false));
```

Comment:
Why casting a boolean to `T`? Can you explain, please?

Reply:
T is a supposed to be a datatype which supports vectorization, something like int. Is there anything wrong with explicitly typecasting false into T?

Reply:
Why not use `0` (zero) instead? or even just `T()`?

Reply:
While writing the test, I was thinking about finding the first true value in the vector. I don't see any difference in behaviour in having false vs 0. [Will change it to 0](https://github.com/STEllAR-GROUP/hpx/pull/6302/commits/adfd8b6eda0d9c4ed5c36ead9f04a44932d22947).

Reply:
FWIW, writing `std::vector<T> v(length);` would do exactly what you want. It would call the default constructor for all new elements, i.e. initializing them to zero as all default constructed builtin types are zero initialized.